docs/contributing/github_integration.rst

```diff
@@ -0,0 +1,115 @@
+
+- **URLs-checker** - buildtest is a GitHub action called **URLs-checker** found at https://github.com/marketplace/actions/urls-checker. The workflow is defined in `urlchecker.yml <https://github.com/HPC-buildtest/buildtest-framework/blob/devel/.github/workflows/urlchecker.yml>`_
+
+.. _black_hook:
+
+Configuring Black Pre-Commit Hook
+-----------------------------------
+
+The above actions check formatting, but are conservative and do not do commits to fix issues on behalf of the user.
+To support an easier workflow, we have provided a git hook that you can install locally to run black directly before each
+commit. To install the hook, simply copy the file to the ``.git/hooks`` folder as follows::
+
+    cp .github/hooks/pre-commit .git/hooks/
+
+
+This hook will exit on error either if you don't have black installed::
+
+    pip install black==19.3b0
+
+
+or if you have black installed, but running it on the repository code results in an error due
+to a functional issue with the code. Code that simply needs to be formatted will be formatted,
+and then the commit will follow.
+
+Once you have installed the ``pre-commit`` hook and black, then you can expect
+black will auto-format your code during the commit phase. Here is a snapshot of
+what the pre-commit hook will do ::
+
+    $ git commit -m "test black on one of the regtest"
+    Black is installed
+    reformatted /mxg-hpc/users/ssi29/buildtest-framework/tests/test_inspect.py
+    All done! ✨ 🍰 ✨
+    1 file reformatted, 39 files left unchanged.
+    [test_black_hook 008fc62] test black on one of the regtest
+     1 file changed, 1 insertion(+)
+
+The pre-commit hook will auto-format specific directories where python files are found. Refer to the
+pre-commit hook (``.github/hooks/pre-commit``) for more details.
+
+If you are interested in running black locally to see diff result from black without auto-formatting code,
+you can do the following::
+
+    $ black --check --diff .
+    --- tests/test_inspect.py       2020-02-25 18:58:58.360360 +0000
+    +++ tests/test_inspect.py       2020-02-25 18:59:07.336414 +0000
+    @@ -18,11 +18,11 @@
+     def test_distro_short():
+         assert "rhel" == distro_short("Red Hat Enterprise Linux Server")
+         assert "centos" == distro_short("CentOS")
+         assert "suse" == distro_short("SUSE Linux Enterprise Server")
+    -    x=0+1*3
+    +    x = 0 + 1 * 3
+
+The changes will be shown with lines removed or added via ``-`` and ``+``. For more details refer to black
+`documentation <https://github.com/psf/black>`_.
+
+GitHub Bots
+-----------
+
+Buildtest has a few bots to do various operations that are described below.
```

Comment:
One thing to bring up for general discussion - I've seen buildtest with upper and lower case B, is the convention to always use lowercase unless we are starting a sentence?

Reply:
Good point, i guess i can change it to lower case to be consistent even if grammatically it may make no sense. The name is `buildtest` so it may be confusing with `Buildtest`

Reply:
Great! Yeah as long as we are consistent we shouldn't see derivations popping up like `Buildtest` or even `BuildTest`.